.azure/modules/postgreSql/create.bicep

```diff
@@ -136,6 +139,46 @@ resource pgms_wait_sampling_query_capture_mode 'Microsoft.DBforPostgreSQL/flexib
   }
 }
 
+resource appInsightsWorkspace 'Microsoft.OperationalInsights/workspaces@2023-09-01' existing = {
+  name: appInsightWorkspaceName
+}
+
+// todo: setting as 0 for now. Will use the log analytics workspace policy instead. Consider setting explicitly in the future.
+var diagnosticSettingRetentionPolicy = {
+  days: 0
+  enabled: false
+}
+
+var diagnosticLogCategories = [
+  'PostgreSQLLogs'
+  'PostgreSQLFlexSessions'
+  'PostgreSQLFlexQueryStoreRuntime'
+  'PostgreSQLFlexQueryStoreWaitStats'
+  'PostgreSQLFlexTableStats'
+  'PostgreSQLFlexDatabaseXacts'
```

Comment:
FYI:

### Diagnostic Log Categories for Azure PostgreSQL Flexible Server

#### PostgreSQLLogs
- Contains general PostgreSQL server logs.
- Includes error messages, authentication attempts, and general server events.
- Essential for troubleshooting server-level issues and security monitoring.

#### PostgreSQLFlexSessions
- Tracks active database sessions and connections.
- Helps monitor concurrent users and connection patterns.
- Useful for identifying connection leaks or excessive concurrent connections.

#### PostgreSQLFlexQueryStoreRuntime
- Captures query execution statistics and runtime metrics.
- Shows query execution times, CPU usage, and I/O statistics.
- Critical for performance tuning and identifying slow queries.

#### PostgreSQLFlexQueryStoreWaitStats
- Records what queries are waiting for and why.
- Helps identify bottlenecks like I/O, locks, or resource contention.
- Essential for diagnosing performance issues and resource conflicts.

#### PostgreSQLFlexTableStats
- Provides statistics about table usage and operations.
- Includes information about table scans, row counts, and index usage.
- Useful for capacity planning and database optimization.

#### PostgreSQLFlexDatabaseXacts
- Monitors database transactions.
- Tracks transaction rates, durations, and commit/rollback patterns.
- Important for understanding database workload and transaction health.